### ISSUE TITLE
Generalization to all resources?

### ISSUE THREAD

**Issue by ghost** [tc39/proposal-oom-fails-fast#2]
It may be better to generalize this to include more than just memory.

Example: right now, the spec has no hard limit on how many Workers an agent may host, and this could define what occurs when the client is out of threads and is unable to execute the workers concurrently anymore.

**Comment by erights** [comment]
I agree. This proposal should at least discuss those cases and take a general normative stance. Thanks.
